### DISCUSSION TITLE
How to put a search field in my sitemap?

### DISCUSSION THREAD

**Discussion by ghost** [datenstrom/community#875] [Ask a question] [answered]
Hi. 
How can I place search field in my sitemap page? Preferably before the display of sitemap content.

I think it should be placed right before:  

`<?php $pages = $this->yellow->page->getPages("sitemap")->paginate($this->yellow->system->get("sitemapPaginationLimit")) ?>`

but I am not a coder, so any help would be great. How do I "invoke" search field in my sitemap? Thank you.

**Comment by GiovanniSalmeri** [comment]
Hello!

First of all: could you please edit the Title of your question and write there the text of the question? (i.e. "How can I place search field in my sitemap page?"). You should use the button Edit at the top right corner. Otherwise it is not clear from the title what you are asking for and the whole discussion risks being flagged and deleted...

Regarding your question, it should suffice to add in the layout this line:

```
<?php echo $this->yellow->page->parseContentShortcut("search", "", "block"); ?>
```

This is equivalent to adding a `[search]` shortcut in a md page. Should the search require a different location (for example `/en/search/`) write it inside the quotes as the second argument. Hope this helps!

Edit: I had written `null` instead of `""`, but this would trigger almost certainly a deprecation warning in PHP 8.

**Reply by ghost** [reply]
Hi again. One year later (and update to 0.9) and now there is an error on the website showing: "Datenstrom Yellow stopped with fatal error. Activate the debug mode for more information." 

I have tried both with "" and "null". 

Debug says: 
`YellowCore->request() #10 {main} thrown in /home/website/system/layouts/default.html on line 6`

The line 6 in `/system/layouts/default.html` is:  
```
<div align="right">
<?php echo $this->yellow->page->parseContentShortcut("search", "", "block"); ?>
</div> 
```



Any suggestion how to fix this?

**Reply by schulle4u** [reply, marked as answer]
The shortcut parsing event has been renamed and modified in the latest API. Please replace the line: 

`<?php echo $this->yellow->page->parseContentShortcut("search", "", "block"); ?>`

With: 

`<?php echo $this->yellow->page->parseContentElement("search", "", "", "block"); ?>`

See also: datenstrom/yellow#967 and [Yellow parse events](https://datenstrom.se/yellow/help/api-for-developers#yellow-parse-events) in the API documentation.

**Reply by ghost** [reply]
Thank you so much. It works now again. 

But just a quick question: I was on 0.8.23 for a long time, and I noticed that there were updates for months. And then I finally discovered there's a new 0.9 version (from April). :) So, because of the new implementations it is not possible to upgrade from 0.8 to 0.9, right? 
I manually installed all my extensions in 0.9. I hope I didn't forget anything before I nuke my current 0.8.23 version site.

Thank you, once again.

**Reply by annaesvensson** [reply]
Thank you for your feedback and good to hear your website is working again. We want to know what people do and what problems they have. Working together and understanding what's going on is essential to make good applications. Let's stick to one topic per discussion, you can write a comment in the [existing release notes](https://github.com/datenstrom/yellow/discussions/categories/see-what-s-new?discussions_q=category%3A%22See+what%27s+new%22+sort%3Adate_created+datenstrom+yellow+released) or [start a new discussion for each question](https://github.com/datenstrom/yellow/discussions/757).... whatever you think fits best..... now you know the drill. 😄

**Comment by ghost** [comment]
Thank you, man. That worked. I changed the name of the question.
